internal/report/source.go

```diff
@@ -214,14 +215,73 @@ func PrintWebList(w io.Writer, rpt *Report, obj plugin.ObjTool, maxFiles int) er
 		}
 
 		printFunctionHeader(w, ff.functionName, path, n.Flat, n.Cum, rpt)
+		percentiles := calculatePercentiles(fnodes)
 		for _, fn := range fnodes {
-			printFunctionSourceLine(w, fn, asm[fn.Info.Lineno], reader, rpt)
+			printFunctionSourceLine(w, fn, asm[fn.Info.Lineno], reader, getPercentileString(float64(fn.Cum), percentiles), rpt)
 		}
 		printFunctionClosing(w)
 	}
 	return nil
 }
 
+func getPercentileString(cumSum float64, percentiles map[float64]float64) string {
+	if cumSum == 0 {
+		return ""
+	}
+	switch {
+	case cumSum >= percentiles[80]:
+		return " percentile_80"
+	case cumSum >= percentiles[60]:
+		return " percentile_60"
+	case cumSum >= percentiles[40]:
+		return " percentile_40"
+	case cumSum >= percentiles[20]:
+		return " percentile_20"
+	case cumSum >= percentiles[10]:
+		return " percentile_10"
+	}
+	return ""
+}
+
+// calculate percentile expects cumSums to be sorted and
+// to contain unique elements. It also expects the percentile to be between 0 and 99.
```

Comment:
It depends on how you define percentile. 100th ptile would mean that the value is bigger than any value in the list, thus it would not exist in the list (this is my understading of ptile, correct me if I'm wrong).